shell/platform/linux/fl_engine.cc

```diff
@@ -603,6 +630,83 @@ FlutterEngineProcTable* fl_engine_get_embedder_api(FlEngine* self) {
   return &(self->embedder_api);
 }
 
+FlutterViewId fl_engine_add_view(FlEngine* self,
+                                 size_t width,
+                                 size_t height,
+                                 double pixel_ratio,
+                                 GCancellable* cancellable,
+                                 GAsyncReadyCallback callback,
+                                 gpointer user_data) {
```

Comment:
FYI: adding/removing a view should block the platform thread until the engine completes the operation successfully. This ensures that the engine's and embedder's states are synchronized so that subsequent view updates (like sending window metric updates or pointer events) and view presents succeed.

For more information on this problem, see: https://flutter.dev/go/multi-view-sync-over-async

I'd consider adding a TODO so we don't forget to fix this later.

Reply:
I've changed the asynchronous call to return the ID after the call completes - this should mean the app can't do anything until this is done. I would expect we would only create the GtkWindow at this point so no events would occur during the call. Does this sound correct @loic-sharma?

Reply:
I'll assume this works, it can be updated later if not - the API is not public.